src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -18,29 +18,68 @@
 #include <app/clusters/bindings/BindingManager.h>
 #include <app/util/binding-table.h>
 
+namespace {
+
+chip::PeerId PeerIdForNode(chip::FabricTable & fabricTable, chip::FabricIndex fabric, chip::NodeId node)
+{
+    chip::FabricInfo * fabricInfo = fabricTable.FindFabricWithIndex(fabric);
+    if (fabricInfo == nullptr)
+    {
+        return chip::PeerId();
+    }
+    return fabricInfo->GetPeerIdForNode(node);
+}
+
+} // namespace
+
 namespace chip {
 
 BindingManager BindingManager::sBindingManager;
 
+CHIP_ERROR BindingManager::UnicastBindingCreated(uint8_t bindingEntryId)
+{
+    EmberBindingTableEntry entry{};
+    emberGetBinding(bindingEntryId, &entry);
+    if (entry.type != EMBER_UNICAST_BINDING)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    return EstablishConnection(entry.fabricIndex, entry.nodeId);
+}
+
+CHIP_ERROR BindingManager::UnicastBindingRemoved(uint8_t bindingEntryId)
+{
+    EmberBindingTableEntry entry{};
+    emberGetBinding(bindingEntryId, &entry);
+    if (entry.type != EMBER_UNICAST_BINDING)
```

Comment:
Is this cross-check that the caller did not mess up worth doing?  I would just assume that if it's claiming to have removed a unicast binding then it did....